pkgs/development/compilers/tinygo/clang-env.patch

```diff
@@ -0,0 +1,47 @@
+diff --git builder/builtins.go builder/builtins.go
+index 76ef3975..2b254a89 100644
+--- builder/builtins.go
++++ builder/builtins.go
+@@ -159,7 +159,7 @@ var aeabiBuiltins = []string{
+ var CompilerRT = Library{
+ 	name: "compiler-rt",
+ 	cflags: func(target, headerPath string) []string {
+-		return []string{"-Werror", "-Wall", "-std=c11", "-nostdlibinc"}
++		return []string{"-Werror", "-Wall", "-std=c11", "-isystem", "@libclang_inc@"}
+ 	},
+ 	sourceDir: "lib/compiler-rt/lib/builtins",
+ 	librarySources: func(target string) []string {
+diff --git builder/picolibc.go builder/picolibc.go
+index af34f7ef..e2c55407 100644
+--- builder/picolibc.go
++++ builder/picolibc.go
+@@ -27,7 +27,7 @@ var Picolibc = Library{
+ 			"-D_COMPILING_NEWLIB",
+ 			"-DHAVE_ALIAS_ATTRIBUTE",
+ 			"-DTINY_STDIO",
+-			"-nostdlibinc",
++			"-isystem", "@libclang_inc@",
+ 			"-isystem", picolibcDir + "/include",
+ 			"-I" + picolibcDir + "/tinystdio",
+ 			"-I" + headerPath,
+diff --git compileopts/config.go compileopts/config.go
+index f33a4b1a..1dfaefc7 100644
+--- compileopts/config.go
++++ compileopts/config.go
+@@ -231,6 +231,8 @@ func (c *Config) CFlags() []string {
+ 		path, _ := c.LibcPath("picolibc")
+ 		cflags = append(cflags,
+ 			"--sysroot="+path,
++			"-isystem", "@libclang_inc@",
++			"-isystem", filepath.Join(path, "include"),
```

Comment:
I don't know why `-isystem` is required for a path added as `--sysroot` the line before. It's not needed on other platforms, and it's my understanding that `clang` adds include paths for sysroots automatically. The build fails with a missing header file if not added.

Reply:
Added upstream.